### DISCUSSION TITLE
Combination of a Vectorstore with a ChatBot

### DISCUSSION THREAD

**Discussion by BeatArnet** [langflow-ai/langflow#322] [Q&A]
I would like to implement a "py-pdf-loader" together with a "ChatBot with memory and prompt". Has anybody made an example to import within langflow?

**Comment by ogabrielluiz** [comment]
Hey @BeatArnet, you can click on the Import button -> Examples and:
<img width="839" alt="image" src="https://github.com/logspace-ai/langflow/assets/24829397/30a1b0e8-a2b1-4dbf-894d-90ea46d74540">

**Comment by BeatArnet** [comment]
Yes, I have already done that. The only thing I'm missing is the combination with a ChatBot that has a memory. The PDF Loader example has no memory. Unfortunately, I can't dock to a previous question. That's why I would like a combination with the Memory ChatBot.

**Reply by ogabrielluiz** [reply]
Oh. I see.. I'll see what I can do.

**Reply by ashmgdesign** [reply]
@BeatArnet Did you have any luck with this?

**Reply by ogabrielluiz** [reply]
I have yet to find an easy way of doing this in langflow. What I thought was to create a new VectorStoreAgent node. Do you have an example LangChain project for me to reference?

**Reply by BeatArnet** [reply]
Unfortunately no. I can find several Langchain examples to process and query a PDF or a text file, almost all of them are called chatbots, but none of them has a memory.

**Comment by carlosrcoelho** [comment]
Hey!

Thanks for sharing your question here. I'm sorry for not getting back to you sooner.
Langflow has gone through some major changes since then. I recommend trying out the new version for the best experience. If you have any issues, please open a new one; we'll be here to help.
